driver/pace/driver/run.py

```diff
@@ -508,10 +587,31 @@ def step_all(self):
             self._step(timestep=self.config.timestep.total_seconds())
             time += self.config.timestep
             self.diagnostics.store(time=time, state=self.state)
+        self.performance_config.total_timer.stop("total")
+        self.performance_config.collect_performance(
+            self.performance_config.timestep_timer,
+            self.performance_config.times_per_step,
+            self.performance_config.hits_per_step,
+        )
+        self.performance_config.write_out_performance(
+            self.comm,
+            self.config.stencil_config.backend,
+            self.config.dt_atmos,
+            self.performance_config.total_timer,
+            self.performance_config.times_per_step,
+            self.performance_config.hits_per_step,
+        )
```

Comment:
Just realized - I wouldn't expect `step_all` to write out performance. Can this be moved to another method, invoked from the top level?

Very optional: If you do this, you could also invoke the timer.start("total") and timer.stop("total") (or use `with`) from the top level `main` function, which would be safer than invoking it in two different methods. I think deleting the "total" times is also an option, I don't see the use case if we already have an initialization + call timing.

Is it an issue that we're timing the first timestep in the same place as the rest? I think we do some compilation at that time.

Reply:
Moved write out performance to be invoked from `main`. Total time is now moved to `main`, but it is the time it takes for `step_all`. I'm also good with just deleting this completely, let's see what @twicki says.